### DISCUSSION TITLE
osxphotos powered website

### DISCUSSION THREAD

**Discussion by eecue** [RhetTbull/osxphotos#1000] [Show and tell]
I've spend my very limited free time over the last few months getting most of my 300k images posted to my website:

https://eecue.com

I used osxphotos extensively: https://eecue.com/blog/extracting-data-from-apple-photos--the-power-of-organization

I used it to extract tags, scores, etc:

Scores: https://eecue.com/indexes/scores-s
Cities: https://eecue.com/indexes/cities-s
Cameras: https://eecue.com/indexes/cameras-s
Lenses: https://eecue.com/indexes/lenses-s

etc... special thanks to @RhetTbull for building this awesome tool.

**Comment by RhetTbull** [comment]
Thanks for sharing @eecue -- looks great!

**Comment by RhetTbull** [comment]
I spent some time exploring this today -- I really like what you've built and how you expose all the metadata in a way that encourages exploration. Thanks again for sharing!

**Reply by oPromessa** [reply]
Yes! Kudos to @RhetTbull !!!

I use `osxphotos` to export into the Synology NAS so that me and the family can access (privately) all the files on the go (with browsers, mobile apps and SmartTV apps) via [Synology Photos](https://www.synology.com/en-global/dsm/feature/photos).

But looking at your site @eecue, it brings different capabilities like searching for "Photos | Scores | Pleasants symmetry" and more.

Did you publish the code to generate the website?

**Comment by eecue** [comment]
I haven’t posted anything yet, but it’s something I’m working towards. On Feb 26, 2023, at 10:16 AM, oPromessa ***@***.***> wrote:﻿
Yes! Kudos to @RhetTbull !!!
I use osxphotos to export into the Synology NAS so that me and the family can access (privately) all the files on the go (with browsers, mobile apps and SmartTV apps) via Synology Photos.
But looking at your site @eecue, it brings different capabilities like searching for "Photos | Scores | Pleasants symmetry" and more.
Did you publish the code to generate the website?

**Comment by RhetTbull** [comment]
I just noticed that this post is issue #1000 and it was posted during the same week osxphotos hit 1000 Github stars.  It's fun to see how far osxphotos has come in 4 years since I first published it as a single script that read keywords and persons from the Photos library!

**Comment by yannikr** [comment]
@eecue Been fascinated by this post for a couple months now and finally requested beta access!

A potential upcoming use case for this, as well as a question:

I will be attending a wedding in early fall where I think it would be a great idea to have a computer/projector setup where guests can see photos of themselves with other people.  Guests could type in "John Doe, Jane Doe" (or even more than 2 people) and see all the photos of them tagged together on a big screen.  This would be taking from a curated library that could end up being a few thousand photos.

Question:

**Is there any current ability to **AND** query for multiple faces/tags?  Such as:**
- Only show photos that are tagged with both **Beach** and **Frisbee**?
- Only show photos that have both **John** and **Jane** in them?

The photos app current allows us to extensively combine queries like "contains text and taken with a Sony camera", but in this case I would only be looking for something that can combine tags.  If faces are not possible, then I could always use osxphotos to bulk tag a photo with whoever's face is tagged.

This would all be running locally with the Mac acting as the "web server".

Not sure if your project would be the best way to go about this, or if what I really need is just a stripped down "display/POS" interface sitting in front of the apple photos app.

Thanks,

Yannik

**Reply by eecue** [reply]
honestly you're probably just best off using Photos for this directly, it works exactly like that
